platform/viewer/src/connectedComponents/ViewerMain.js

```diff
@@ -65,63 +63,75 @@ class ViewerMain extends Component {
       this.setState({
         displaySets,
       });
+
+      this.fillEmptyViewportPanes();
     }
   }
 
   componentDidUpdate(prevProps) {
-    if (this.props.studies !== prevProps.studies) {
+    const prevViewportAmount = prevProps.layout.viewports.length;
+    const viewportAmount = this.props.layout.viewports.length;
+    const isVtk = this.props.layout.viewports.some(vp => !!vp.vtk);
```

Comment:
So if any `layout.viewports` has a `vtk` property (what we cram our public API on to)
Then DO NOT `fillEmptyViewportPanes`

That's interesting. I think I get the gist of why, but it's a bit muddy.
If this works, great :+1:
I think we'll still have a fair bit of clean-up we can do as we iterate on this layout logic.